src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,161 @@
+package duke;
+
+
+
+import java.util.List;
+
+/**
+ * Represents the user interface for the Duke program.
+ */
+public class Ui {
+    public Ui() {
+
+    }
+
+    /**
+     * Displays a message asking the user to specify the content of a todo list item.
+     */
+    public void specify() {
+        System.out.println("Please specify the content of the todo list");
+    }
+
+    /**
+     * Displays a message indicating that a task has been removed.
+     *
+     * @param task The removed task.
+     */
+    public void remove(String task) {
+        System.out.println("Noted. I've removed this task:");
+        System.out.println(task);
+    }
+
+    /**
+     * Displays a message asking the user to input an integer to select a task.
+     */
+    public void numExc() {
+        System.out.println("to pick which task to do, please input an integer");
+    }
+
+    /**
+     * Displays a message indicating that the specified task index does not exist in the task list.
+     */
+    public void indexOut() {
+        System.out.println("currently, your task list does not contain the task with the index you just inputted");
+    }
+
+    /**
+     * Displays the list of tasks.
+     *
+     * @param list The list of tasks to display.
+     */
+    public void list(List<Task> list) {
+        int index = 1;
+        System.out.println("Here are the tasks in your list:");
+        System.out.println("____________________________________________________________");
+        for (Task thing: list) {
+            System.out.println(index +". " + thing.toString());
+            index++;
+        }
+        System.out.println("____________________________________________________________");
+    }
+
+    /**
+     * Displays the matching list of tasks.
+     *
+     * @param list The matching list of tasks to display.
+     */
+    public void matchingList(List<Task> list) {
```

Comment:
Perhaps it might be better to name this in the verb form? e.g displayMathchingList